hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/StandaloneEncryptedFileSystem.java

```diff
@@ -139,6 +139,18 @@ public boolean delete(Path path, boolean recursive) throws IOException {
         return delegate.delete(path, recursive);
     }
 
+    @Override
+    public boolean rename(Path src, Path dst) throws IOException {
+        // The EncryptedFileSystem only works with files to allow the KeyStorageStrategy interface to be simple.
+        // Because we use the FileKeyStorageStrategy we know that we can also support operations on directories
+        // and therefore delegate to the underlying FileSystem in that case.
+        if (fs.isFile(src)) {
+            return fs.rename(src, dst);
+        } else {
+            return delegate.rename(src, dst);
```

Comment:
@uschi2000 how do you feel about this logic and the comment? think it's clear enough and can replace what I've got in the `delete` implementation.

Reply:
I would make the leap from "operations on directories" to "can delegate" more explicit: Since StandaloneEncryptedFileSystem uses a FileKeyStorageStrategy, the delegate delete/rename call on folders deletes both the payload files and the adjacent encryption materials. For files ...

remember, you're not writing the docs for yourself, but for someone else who potential has no clue how this things works.